crystalfontz/dbus/client.py

```diff
@@ -0,0 +1,548 @@
+    await client.set_line_2(line.encode("utf-8"), TimeoutM.none, RetryTimesM.none)
+
+
+@main.command(help="Interact with special characters")
+def character() -> None:
+    raise NotImplementedError("crystalfontzctl character")
+
+
+@main.group(help="Interact directly with the LCD controller")
+def lcd() -> None:
+    pass
+
+
+@lcd.command(name="poke", help="10 (0x0A): Read 8 Bytes of LCD Memory")
+@click.argument("address", type=BYTE)
+@async_command
+@pass_client
+async def read_lcd_memory(client: DbusClient, address: int) -> None:
+    memory = await client.read_lcd_memory(address, TimeoutM.none, RetryTimesM.none)
+    echo(LcdMemoryM.load(memory))
+
+
+@main.group(help="Interact with the LCD cursor")
+def cursor() -> None:
+    pass
+
+
+@cursor.command(name="position", help="11 (0x0B): Set LCD Cursor Position")
+@click.argument("row", type=BYTE)
+@click.argument("column", type=BYTE)
+@async_command
+@pass_client
+async def set_cursor_position(client: DbusClient, row: int, column: int) -> None:
+    await client.set_cursor_position(row, column, TimeoutM.none, RetryTimesM.none)
+
+
+@cursor.command(name="style", help="12 (0x0C): Set LCD Cursor Style")
+@click.argument("style", type=click.Choice([e.name for e in CursorStyle]))
+@async_command
+@pass_client
+async def set_cursor_style(client: DbusClient, style: str) -> None:
+    await client.set_cursor_style(
+        CursorStyleM.dump(CursorStyle[style]), TimeoutM.none, RetryTimesM.none
+    )
+
+
+@main.command(help="13 (0x0D): Set LCD Contrast")
+@click.argument("contrast", type=float)
+@async_command
+@pass_client
+async def contrast(client: DbusClient, contrast: float) -> None:
+    await client.set_contrast(contrast, TimeoutM.none, RetryTimesM.none)
+
+
+@main.command(help="14 (0x0E): Set LCD & Keypad Backlight")
+@click.argument("brightness", type=float)
+@click.option("--keypad", type=float)
+@async_command
+@pass_client
+async def backlight(
+    client: DbusClient, brightness: float, keypad: Optional[float]
+) -> None:
+    await client.set_backlight(
+        brightness, OptFloatM.dump(keypad), TimeoutM.none, RetryTimesM.none
+    )
+
+
+@main.group(help="DOW (Dallas One-Wire) capabilities")
+def dow() -> None:
+    pass
+
+
+@dow.command(name="info", help="18 (0x12): Read DOW Device Information")
+@click.argument("index", type=BYTE)
+@async_command
+@pass_client
+async def read_dow_device_information(client: DbusClient, index: int) -> None:
+    info = await client.read_dow_device_information(
+        index, TimeoutM.none, RetryTimesM.none
+    )
+    echo(DowDeviceInformationM.load(info))
+
+
+@main.group(help="Temperature reporting and live display")
+def temperature() -> None:
+    pass
+
+
+@temperature.command(name="reporting", help="19 (0x13): Set Up Temperature Reporting")
+@click.argument("enabled", nargs=-1)
+@async_command
+@pass_client
+async def setup_temperature_reporting(client: DbusClient, enabled: Tuple[int]) -> None:
```

Comment:
Is this Tuple type correct? I'd expect a list, or at least a type `Tuple[int, ...]`.

Reply:
Captured in #66 .

Reply:
I pushed code in this branch to fix. I'll need to QA, but I'll be QAing the branch anyway so I'm not super concerned.